staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go

```diff
@@ -26,28 +26,38 @@ import (
 	restclient "k8s.io/client-go/rest"
 )
 
-func (s *SecureServingInfo) NewSelfClientConfig(token string) (*restclient.Config, error) {
-	if s == nil || (s.Cert == nil && len(s.SNICerts) == 0) {
-		return nil, nil
-	}
-
-	host, port, err := net.SplitHostPort(s.ServingInfo.BindAddress)
+// loopbackBindAddress determines the address to query when constructing a
+// loopback REST client.
+func loopbackBindAddress(bindAddress string) string {
```

Comment:
It's not really the BindAddress, but the address to connect to derived from the BindAddress. So `loopbackHostForBindAddress` or something like that.